File: lib/decrypt.rb
require 'json'

def decrypt_text(decrypt_info)
  decrypt_info = decrypt_info.to_json
  `python3 #{Dir.pwd}/decrypt.py '#{decrypt_info}'`
end

def decrypt_text_from_env
  return {} unless ENV['text_encrypt_ascii']

  decrypt_info = {
	  "encrypt_key" => ENV['encrypt_key'],
	  "work_key_encrypt" => ENV['work_key_encrypt'],
	  "work_key_encrypt_iv" => ENV['work_key_encrypt_iv'],
	  "text_iv_ascii" => ENV['text_iv_ascii'],
	  "text_encrypt_ascii" => ENV['text_encrypt_ascii']
  }
  text = decrypt_text(decrypt_info)
  return JSON.parse(text)
rescue StandardError => e
  puts e
  return {}
end

File: sbin/decrypt.py
import os
import sys
import json

sys.path.append(os.path.abspath("../"))
from lib.crypt_env import decrypt_text

try:
    decrypt_info_json = sys.argv[-1].replace("'", '"')
    decrypt_info = json.loads(decrypt_info_json)
    text = decrypt_text(
        decrypt_key=decrypt_info["encrypt_key"],
        work_key_encrypt=decrypt_info["work_key_encrypt"],
        work_key_encrypt_iv=decrypt_info["work_key_encrypt_iv"],
        text_iv_ascii=decrypt_info["text_iv_ascii"],
        text_encrypt_ascii=decrypt_info["text_encrypt_ascii"]
    )
    sys.stdout.write(text)
    sys.stdout.flush()
except Exception as e:
    print(e)
